src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -422,6 +424,15 @@ private static NamedTypeSymbol ValidateBuilderType(SyntheticBoundNodeFactory F,
                     !property.IsStatic &&
                     (property.ParameterCount == 0))
                 {
+                    if (property.Type != returnType)
```

Comment:
Discussed this with Chuck. You're correct there is an inconsistency: we consider tuple names and dynamic-ness, but ignore custom modifiers. That seems strange.
I see two ways out of this:

1. we should add a check for custom modifiers (this seems like an existing bug, although low-pri)
2. we should relax the check and ignore tuple/dynamic differences

I'm tempted to go with (1) from a language perspective, because I can't think of any reason someone would write a task-like type and its corresponding async method builder with such differences. Would that make sense?